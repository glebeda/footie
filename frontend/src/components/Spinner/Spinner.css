.spinner {
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 20px;
  }
  
  .spinner__circle {
    width: 40px;
    height: 40px;
    border: 4px solid #ccc;
    border-top-color: #1d70b8; 
    border-radius: 50%;
    animation: spin 0.8s linear infinite;
  }
  
  @keyframes spin {
    to {
      transform: rotate(360deg);
    }
  }
  