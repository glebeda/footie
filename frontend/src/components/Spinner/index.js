import React from 'react';
import './Spinner.css'; 

const Spinner = () => {
  return (
    <div className="spinner">
      <div className="spinner__circle" aria-label="Loading"></div>
    </div>
  );
};

export default Spinner;
