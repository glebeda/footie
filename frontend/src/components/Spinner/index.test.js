import React from 'react';
import { render, screen } from '@testing-library/react';
import Spinner from './index';

describe('Spinner', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Spinner />);
    expect(asFragment()).toMatchSnapshot();

    // Check that the spinner is in the document
    expect(screen.getByLabelText('Loading')).toBeInTheDocument();
  });
});
