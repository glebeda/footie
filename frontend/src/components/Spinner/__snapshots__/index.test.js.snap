// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders correctly 1`] = `
<DocumentFragment>
  <div
    class="spinner"
  >
    <div
      aria-label="Loading"
      class="spinner__circle"
    />
  </div>
</DocumentFragment>
`;
